Add `from json(string): Any` conversion function

npepinpe opened the pull request:
## Description

This PR adds a new `from json(string): Any` built-in conversion function, allowing users to convert stringified-JSON results (very common with agentic orchestration) into objects, such as contexts, lists, strings, etc. This simplifies extracting, transforming, etc., parts from such results.

The tests were largely taken [from this comment](https://github.com/camunda/feel-scala/issues/602#issuecomment-1591197695), major thanks for this :pray: 

One small issue, Jackson requires an additional compiler file, `Ytasty-reader`. I'm a complete Scala noob, so while I could add it and make progress, I cannot speak for its impact and if that's something we want to move forward with: does it mean downstream users also need to add this flag? If so, do we want that?

## Related issues

closes #825

npepinpe commented:
Happy to implement #602 if this is merged. I expect the complexity is similar, and these kind of things are all I can really manage nowadays :(

saig0 reviewed (changes requested):
@npepinpe thank you for your contribution. :tada: Your Scala skill increased by 3 EXP. :muscle: 

It looks good. I added a few minor comments. :cookie:

npepinpe commented on pom.xml:
```diff
@@ -177,6 +189,9 @@
               <goal>add-source</goal>
               <goal>compile</goal>
             </goals>
+            <configuration>
+              <addScalacArgs>-Ytasty-reader</addScalacArgs>
+            </configuration>
```
See, this is what happens because I don't know the Scala ecosystem at all, and I really just wanted my function to parse JSON :smile:

npepinpe commented on src/main/scala/org/camunda/feel/impl/builtin/ConversionBuiltinFunctions.scala:
```diff
@@ -301,6 +308,21 @@ class ConversionBuiltinFunctions(valueMapper: ValueMapper) {
+          .map { value: Any =>
+            valueMapper.toVal(value)
+          }
+          .getOrElse {
+            ValError(s"Failed to parse JSON from '${json.value}'")
```
Can you elaborate what you mean by return the error _first_, and then the JSON? Is there an example of something similar? Like, return a tuple?

npepinpe commented on src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala:
```diff
@@ -528,4 +528,54 @@ class BuiltinConversionFunctionsTest
     ) should returnResult(Period.parse("P1Y8M"))
   }
 
+  "A from json(string) function" should "convert a JSON string" in {
```
The contribution guide is not very clear though:

```
Add a new test case following the pattern: "A reverse() function" should "return a string in reverse order" in { ??? }
Add more test cases for other results, alternative parameters, or covering edge cases
```

But also

```
Group all test cases for a method using it should "return null if the argument is not a string" in { ??? }
```

What is the criteria for splitting test cases? Input? Output? Both?

npepinpe commented:
Addressed most of the review feedback, thanks! Just wasn't sure about two points:

- The test split. It's still unclear to me, even from the contributing guide, if the split is over input, output, both, or...? The test file itself doesn't seem to always be consistent.
- I didn't get the point about the error return, i.e. return the error _first_, then the JSON? What's the two-step return? :thinking:

saig0 reviewed (approved):
Nice. :+1:  I grouped the test cases.  

It is ready to merge. :rocket:
